apifederator/pkg/apis/apifederation/types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apifederation
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// APIServerList is a list of APIServer objects.
+type APIServerList struct {
+	unversioned.TypeMeta
+	unversioned.ListMeta
+
+	Items []APIServer
+}
+
+// APIServerSpec contains information for locating and communicating with a server.
+// Only https is supported, though you are able to disable certificate verification.
+type APIServerSpec struct {
+	// InternalHost is the host:port for locating the server inside the pod/service network.
+	InternalHost string
+	// Group is the API group name this server hosts
+	Group string
+	// Version is the API version this server hosts
+	Version string
```

Comment:
>> does "side-by-side" mean more than one server could serve the same group/version?
>
> No. One service for v1, another service for v2.

how do you plan to enforce that in case of conflicts? oldest wins?